### PR TITLE
Filter prs with failing required jobs

### PR DESCRIPTION
**Pull request by petr-muller**
Followup to https://github.com/openshift/ci-tools/pull/2675,

Removes a ton of useless Tide stuff and slightly reworks the logic. The deployment will need access to Prow *job* configuration.

/cc @hongkailiu

### CODE REVIEW

**Comment by petr-muller** [comment]
This is a log snippet that I got by running the code with 
```
--job-config-path=/home/afri/Projects/RH/go/src/github.com/openshift/release/ci-operator/jobs
--config-path=/home/afri/Projects/RH/go/src/github.com/openshift/release/core-services/prow/02_config/_config.yaml
--dry-run=true
--github-token-path=/home/afri/Temporary/shift-week-token
--supplemental-prow-config-dir=/home/afri/Projects/RH/go/src/github.com/openshift/release/core-services/prow/02_config
--run-once=true
```

There's a lot of GH client log clutter there but the filtering does work ;)

[retester.log](https://github.com/openshift/ci-tools/files/8074482/retester.log)

File: cmd/better-retester/retester.go [hunk 1]
```diff
@@ -55,27 +55,19 @@ func (c *retestController) sync() error {
 		return err
 	}
 
-	rawPools, err := c.dividePool(candidates)
-	if err != nil {
-		return err
+	logrus.Infof("Found %d candidates for retest (pass label criteria, fail some tests)", len(candidates))
```
Comment (hongkailiu) [inline cmd/better-retester/retester.go:57]: Do we know "fail some tests" here?

File: cmd/better-retester/retester.go [hunk 2]
```diff
@@ -90,6 +82,31 @@ func findCandidates(config config.Getter, gc githubClient, usesGitHubAppsAuth bo
 	return prs, nil
 }
 
+func (c *retestController) atLeastOneRequiredJob(candidates map[string]tide.PullRequest) (map[string]tide.PullRequest, error) {
+	output := map[string]tide.PullRequest{}
+	for key, pr := range candidates {
+		// All
```
Comment (hongkailiu) [inline cmd/better-retester/retester.go]: // All presubmits that are required to pass for merging the PR.

File: cmd/better-retester/retester.go [hunk 3]
```diff
@@ -90,6 +82,31 @@ func findCandidates(config config.Getter, gc githubClient, usesGitHubAppsAuth bo
 	return prs, nil
 }
 
+func (c *retestController) atLeastOneRequiredJob(candidates map[string]tide.PullRequest) (map[string]tide.PullRequest, error) {
```
Comment (hongkailiu) [inline cmd/better-retester/retester.go:89]: This is the function that I failed to extract from tide.
I really tried, but failed.

File: cmd/better-retester/retester.go [hunk 4]
```diff
@@ -249,7 +267,7 @@ func notInBackOff(input map[string]tide.PullRequest) map[string]tide.PullRequest
 	return input
 }
 
-func retest(prs map[string]tide.PullRequest) {
+func retest(gprs map[string]tide.PullRequest) {
```
Comment (hongkailiu) [inline cmd/better-retester/retester.go:270]: what is the `g` here?

**Review by hongkailiu — COMMENTED** [review]
/lgtm

/hold

to give a chance to address my comments.

File: cmd/better-retester/retester.go (hunk 1 above)
Comment (petr-muller) [reply cmd/better-retester/retester.go:57]: Yeah - that's something I added to the `query` method:

File: cmd/better-retester/retester.go (hunk 2 above)
Comment (petr-muller) [reply cmd/better-retester/retester.go]: Yep will address (but tomorrow)

File: cmd/better-retester/retester.go (hunk 4 above)
Comment (petr-muller) [reply cmd/better-retester/retester.go:270]: hehe a mistake

Comment (petr-muller) [reply cmd/better-retester/retester.go:270]: will address

File: cmd/better-retester/retester.go [hunk 5]
```diff
@@ -3,58 +3,76 @@ package main
...
-		logrus.Infof("Candidate PR: %s (https://github.com/%s/%s/pull/%d", key, pr.Repository.Owner.Login, pr.Repository.Name, pr.Number)
+	logrus.Infof("Found %d candidates for retest (pass label criteria, fail some tests)", len(candidates))
+	for _, pr := range candidates {
+		logrus.Infof("Candidate PR: https://github.com/%s/%s/pull/%d", pr.Repository.Owner.Login, pr.Repository.Name, pr.Number)
 	}
 
-	// Input: A list of PRs that would merge but have failing jobs
-	// Output: A subset of input PRs whose jobs are actually required for merge
-	candidates = atLeastOneFailingRequiredJob(candidates)
+	candidates, _ = c.atLeastOneRequiredJob(candidates)
```
Comment (petr-muller) [inline cmd/better-retester/retester.go:62]: I also need to not ignore this error

**Pull request merged**